### PR TITLE
Chore/22626 upgrade nodejs

### PR DESCRIPTION
**Pull request by MStreet3**
upgrades nodejs from 12 to 16LTS.

### CODE REVIEW

**Review by jmank88 — APPROVED** [review]

File: tools/external-adapter/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM node:12.22.6-alpine
+FROM node:16.13.2
```
Comment (jmank88) [inline tools/external-adapter/Dockerfile:1]: Was the alpine variant important here for size?

File: core/chainlink.Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # MAKE ALL CHANGES WITHIN THE DEFAULT WORKDIR FOR YARN AND GO DEP CACHE HITS
-FROM node:12-buster
+FROM node:16
```
Comment (jmank88) [inline core/chainlink.Dockerfile:2]: Same for buster.

**Review by jmank88 — CHANGES_REQUESTED** [review]

File: tools/external-adapter/Dockerfile (hunk 1 above)
Comment (MStreet3) [reply tools/external-adapter/Dockerfile]: using [16-alpine3.15](https://hub.docker.com/layers/node/library/node/16-alpine3.15/images/sha256-0b4d0c4df160b7e9ea5fa8e366e62fc95f0b3a376b4ae4295acc593280330b00?context=explore) image which is `16.13.2` at ~37MB.  versus [16-buster](https://hub.docker.com/layers/node/library/node/16-buster/images/sha256-7c49a64aba86dd483aa874bc0230c07f282e20741a7c66e426970ecafc149a38?context=explore) at ~332MB.

File: core/chainlink.Dockerfile [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # MAKE ALL CHANGES WITHIN THE DEFAULT WORKDIR FOR YARN AND GO DEP CACHE HITS
-FROM node:12-buster
+FROM node:16-alpine3.15
```
[the remark this answers is not in the document]
Comment (MStreet3) [reply core/chainlink.Dockerfile]: so alpine doesn't work here.  I went back to [16-buster](https://hub.docker.com/layers/node/library/node/16-buster/images/sha256-7c49a64aba86dd483aa874bc0230c07f282e20741a7c66e426970ecafc149a38?context=explore) but it has the same build steps/layers as [16](https://hub.docker.com/layers/node/library/node/16/images/sha256-e9d5cd564ae1e9f9a0f3f5f4a0ba2f6df42ecce5d78b317480fa78a1d12539e1?context=explore)

**Review by jmank88 — APPROVED** [review]

**Pull request merged**